detox/src/devices/drivers/android/AndroidDriver.js

```diff
@@ -58,6 +64,16 @@ class AndroidDriver extends DeviceDriverBase {
     };
   }
 
+  /** @protected */
+  async allocateDevice(matcher) {
+    log.debug({ event: 'ALLOCATE_DEVICE' }, `Trying to allocate a device based on %j`, matcher);
+    const adbName = await this.deviceRegistry.allocateDevice(() => {
+      return this.freeDeviceFinder.findFreeDevice(matcher);
```

Comment:
Not very fond of this part -- I don't appreciate the base implementation imposing this logic. Better keep the `freeDeviceFinder` at the implementor's side, even with the cost of (very little) code duplication, if you ask me.
I'm nevertheless okay with the skeletal implementation here, but perhaps with the subtle replacement of `return this.freeDeviceFinder.findFreeDevice(matcher)` with `return this._someAbstractMethod()`

Reply:
I see. @d4vidi, do you mean that `this.freeDeviceFinder = null;` in the base class looks somewhat ugly?
Well, I tend to agree — both implementors of AndroidDriver can handle its initialization completely, without the base `AndroidDriver` class knowing those details, if we make `findFreeDevice` method abstract. 👍

Reply:
yes, exactly